docs/how-to-guides/plugin-sidebar.md

```diff
@@ -0,0 +1,380 @@
+-   a `TextControl` component instead of the plain `'Meta field'` text.
+
+With the new CSS class available you can add a little style. Create a new file in your plugin directory called `plugin-sidebar.css` with the following to give some padding:
+
+```css
+.plugin-sidebar-content {
+	padding: 16px;
+}
+```
+
+Register the script and enqueue it to load with `enqueue_block_editor_assets` alongside the JavaScript file.
+
+After those changes, the PHP code will look like this:
+
+```php
+<?php
+
+/*
+Plugin Name: Sidebar example
+*/
+
+function sidebar_plugin_register() {
+	wp_register_script(
+		'plugin-sidebar-js',
+		plugins_url( 'plugin-sidebar.js', __FILE__ ),
+		array(
+			'wp-plugins',
+			'wp-edit-post',
+			'wp-element',
+			'wp-components'
+		)
+	);
+	wp_register_style(
+		'plugin-sidebar-css',
+		plugins_url( 'plugin-sidebar.css', __FILE__ )
+	);
+}
+add_action( 'init', 'sidebar_plugin_register' );
+
+function sidebar_plugin_script_enqueue() {
+	wp_enqueue_script( 'plugin-sidebar-js' );
+	wp_enqueue_style( 'plugin-sidebar-css' );
+}
+add_action( 'enqueue_block_editor_assets', 'sidebar_plugin_script_enqueue' );
+```
+
+Reload the editor and open the sidebar:
+
+![Sidebar with style and controls](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/assets/sidebar-style-and-controls.png)
+
+This code doesn't let users store or retrieve data just yet, so the next steps will focus on how to connect it to the meta block field.
+
+### Step 3: Register the Meta Field
+
+To work with fields in the `post_meta` table, use the [register_post_meta](https://developer.wordpress.org/reference/functions/register_post_meta/). function to create a new field called `sidebar_plugin_meta_block_field`.
+
+Note: this field needs to be available to the REST API because that's how the block editor access data.
+
+Add the PHP code in your plugins `init` callback function:
+
+```php
+register_post_meta( 'post', 'sidebar_plugin_meta_block_field', array(
+	'show_in_rest' => true,
+	'single' => true,
+	'type' => 'string',
+) );
+```
+
+To confirm, query the block editor store to see the field is loaded . After implementing, reload the editor page and open your browser's developer console.Use this JavaScript snippet in the console to confirm:
+
+```js
+wp.data.select( 'core/editor' ).getCurrentPost().meta;
+```
+
+The function will return an object containing the registered meta field you registered.
+
+If the code returns `undefined` make sure your post type supports `custom-fields`. Either when [registering the post](https://developer.wordpress.org/reference/functions/register_post_type/#supports) or with [add_post_type_support function](https://developer.wordpress.org/reference/functions/add_post_type_support/).
+
+### Step 4: Initialize the Input Control
+
+With the field available in the editor store, it can be surfaced to the UI. We extract the input control to a function to keep the clean as we add functionality.
```

Comment:
```suggestion
With the field available in the editor store, it can now be surfaced to the UI. We extract the input control to a function to keep the code clean as we add functionality.
```